managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -4012,4 +4034,55 @@ public ManagedLedgerInternalStats.CursorStats getCursorStats() {
         cs.properties = getProperties();
         return cs;
     }
+
+    /**
+     * Called by ManagedLedgerImpl to execute the Runnable inside the lock to remove the cursor from it's
+     * waiting cursors list.
+     * The cursor state is set to unregistered, and it can be registered again for waiting in ManagedLedgerImpl.
+     */
+    void removeWaitingCursorRequested(Runnable removeWaitingCursorRunnable) {
+        synchronized (waitingRegistrationLock) {
```

Comment:
This method will be called on the publishing messages path, which is a really hot path, right?

Reply:
There's no calls to `ledger.removeWaitingCursor(this)` in the hot path, so this should be fine.